enterprise/frontend/src/metabase-enterprise/embedding_iframe_sdk_setup/hooks/use-hide-parameter.ts

```diff
@@ -0,0 +1,34 @@
+import { useSdkIframeEmbedSetupContext } from "../context";
+
+export function useHideParameter() {
+  const { settings, updateSettings } = useSdkIframeEmbedSetupContext();
+
+  const toggleParameterVisibility = (parameterName: string) => {
+    // Only dashboards supports hiding parameters
+    if (!settings.dashboardId) {
+      return;
+    }
+
+    const hiddenParameters = settings?.hiddenParameters ?? [];
+
+    const nextHiddenParameters = hiddenParameters.includes(parameterName)
+      ? hiddenParameters.filter((name) => name !== parameterName)
+      : [...hiddenParameters, parameterName];
+
+    updateSettings({ hiddenParameters: nextHiddenParameters });
+  };
+
+  const isParameterHidden = (parameterName: string) => {
+    // Only dashboards support hiding parameters
+    if (!settings.dashboardId) {
+      return false;
```

Comment:
This feels wrong. I know it won't make any difference, but the comment says `Only dashboards support hiding parameters` which implies the questions don't support hiding parameters, which must mean it will always show the parameters right? Wrong!!!! it will always hide the parameters.

We should probably update the comment explicitly calling out the default behavior for both dashboards and questions since they're quite different.

Secondly, we know that by default we don't show the question parameters. I'd assume the `isParameterHidden(questionParam)` should return `true`, but I know this value isn't used so it doesn't make any difference at the moment. But like I said, it just feels wrong.

```suggestion
      return true;
```